app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    func_cache = {}
+
+    def wrapper(*args) -> Any:
+        if args in func_cache:
```

Comment:
Using mutable types like lists or dictionaries as keys in a dictionary will raise a TypeError because they are not hashable. Consider using only immutable types like tuples as keys, or converting the arguments to a hashable type before using them as keys.